odata-protocol/11.4 Data Modification.md

```diff
@@ -148,6 +148,21 @@ to that collection's URL. The `POST` body MUST contain a single valid
 representation of an entity of the declared target entity type,
 or one of its derived types.
 
+If the service is unable to determine the canonical collection (as defined in [section ##ContextURL])
+for the entity, it MUST fail the request.
+This determination is possible, for example, if the collection's URL
+- consists of an entity set
+- ends with a containment navigation property
+- ends with a non-containment navigation property with a navigation property binding
```

Comment:
Really? A navigation property binding
```xml
<EntitySet Name="Orders">
  <NavigationPropertyBinding Path="Items/CancellationItem" Target="ns.container/Orders/Items" />
</EntitySet>
```
is insufficient to determine the canonical collection for the create request `POST Orders(23)/Items(5)/CancellationItem`. This needs service-specific knowledge that the cancellation item belongs to the same order, hence the collection is `Orders(23)/Items`.

Reply:
This is not a valid [binding target](https://docs.oasis-open.org/odata/odata-csdl-xml/v4.02/csd01/odata-csdl-xml-v4.02-csd01.html#BindingTarget). 

> If the target is a target path, it MUST resolve to an entity set, singleton, or direct or indirect containment navigation property of a _singleton_ in scope.

With that restriction in mind line 156 should work.

Reply:
The term "containment navigation property of a singleton" seems undefined. Is this
- a single-valued containment navigation property?
- a path starting with a singleton that ends in a containment navigation property?

Why are singletons treated specially here?

Reply:
> The path can traverse single-valued containment navigation properties or single-valued complex properties before ending in a containment navigation property, and there MUST NOT be any non-containment navigation properties prior to the final segment.

What is missing in this definition?

Reply:
This definition is OK, but the additional condition
> If the target is a target path, it MUST resolve to an entity set, singleton, or direct or indirect _containment navigation property of a singleton_ in scope.

leaves unclear where the singleton must be (at the beginning or the end of the path).

Reply:
Thanks! Please suggest an addition.

Reply:
Discussed 2025-3-26 and agreed that the current wording is sufficient.